examples/retry_command.ts

```diff
@@ -0,0 +1,34 @@
+import { Command, CommandStep, Pipeline } from '../src';
+import { YamlSerializer } from '../src/serializers/yaml';
+
+class RetryCommand extends Command {
+    constructor(public retries: number, public c: Command) {
+        super(c.toString());
+        this.retries = retries;
+    }
+
+    protected serialize(): string {
+        return new Array(this.retries + 1).fill(this.c).join(' || ');
+    }
+
+    public toString(): string {
+        return this.serialize();
+    }
+}
+
+const install = new RetryCommand(1, new Command('yarn'));
+
+const lint = new CommandStep([
+    install,
+    new RetryCommand(1, new Command('yarn lint')),
```

Comment:
```suggestion
    new RetryCommand(1, new Command('yarn test')),
```

or even:
```suggestion
    new RetryCommand(1, new Command('yarn test-integration')),
```

- linting is usually very deterministic.